journals/week5.md

```diff
@@ -0,0 +1,186 @@
+# Week 5: Prototyping and Jobtap 
+
+### Refactoring Fripp Code
+
+Since last week, I made some small changes to the implementation in `fripp.c`. First, I added the socket and the packet buffer to the `fripp_ctx`.
+
+```c
+struct fripp_ctx {
+	flux_t *h;
+	struct sockaddr_in si_server;
+	int sock;
+	char *buf;
+	int buf_len;
+};
+```
+
+In turn, that means that the socket is now opened only once per context (when it's being created in `fripp_ctx_create`) and closed once upon destruction of the context, and there is no longer a fixed-size internal buffer for each send (`fripp_format_send`). Instead the buffer in the context is dynamically sized and resized if needed with the idea that most packets being sent will be the same length for a while. One last notable change is that within `fripp_format_send` there is now a second `va_list` (a copy) that's used when the buffer needs to be resized. This was because sometimes, and I'm not sure why it didn't happen every time, upon resizing the buffer the second `vsnprintf` would segfault.
+
+### Timing Asynchronous Load/Stores from the Content Cache
+
+Last week I began collecting some metrics from the content-cache including timing the asynchronous load/store operations from the content cache. However, I stored the timespec for each load/store in auxiliary storage on the flux handle
+
+```c
+struct  timespec  *t0;
+if (!(t0  =  malloc (sizeof (*t0)))) {
+	flux_log (cache->h, LOG_ALERT, "fripp time start");
+} else {
+	monotime (t0);
+	flux_aux_set (cache->h, "fripp::cc-store", t0, free);
+}
+```
+
+That turned out to be the wrong approach as each load/store could (or more likely would) be overwriting the previously stored starting time since they are all the same name on the single flux handle. So, the timing metrics I showed last week probably don't mean much. To really time the load/store operations, I instead added a timespec to each `cache_entry`, and within the `cache_load` and `cache_store` callbacks set the start time at the beginning of the call using:
+
+```c
+monotime (&e->fripp_timestamp)
+```
+
+and then got and sent the final time within the `cache_load_continuation` and `cache_store_continuation` callbacks like:
+
+```c
+fripp_format_send (cache->fripp, "flux.content-cache.store:%lf|ms",
+		   monotime_since (e->fripp_timestamp));
+```
+
+**Load Percentiles**
```

Comment:
Might be worth mentioning the Y axis units in the timing graphs.  I'm assuming its milliseconds?